com.unity.ml-agents.extensions/Runtime/Sensors/PhysicsBodySensor.cs

```diff
@@ -0,0 +1,93 @@
+using UnityEngine;
+using Unity.MLAgents.Sensors;
+
+namespace Unity.MLAgents.Extensions.Sensors
+{
+    /// <summary>
+    /// ISensor implementation that generates observations for a group of Rigidbodies or ArticulationBodies.
+    /// </summary>
+    public class PhysicsBodySensor : ISensor
+    {
+        int[] m_Shape;
+        string m_SensorName;
+
+        PoseExtractor m_PoseExtractor;
+        PhysicsSensorSettings m_Settings;
+
+        /// <summary>
+        ///  Construct a new PhysicsBodySensor
+        /// </summary>
+        /// <param name="rootBody"></param>
+        /// <param name="settings"></param>
+        /// <param name="sensorName"></param>
+        public PhysicsBodySensor(Rigidbody rootBody, PhysicsSensorSettings settings, string sensorName=null)
```

Comment:
I might need to rethink this a bit, since for Worm prefab, the GameObjects that have rigid bodies are all at the same level, so the current RigidBodyPoseExtractor doesn't find the other bodies.

Reply:
We could have a constructor that takes an array or Rigidbody as argument. This should help for the worm case.

Reply:
Yeah, I think we're going to need that eventually, so that we can exclude bodies we don't care about (for example, the eyes on the Worm). It would be nice to pre-populate the list from a root though; need to experiment with that in the editor eventually.